src/main/java/com/autotune/analyzer/serviceObjects/verification/validators/KubernetesElementsValidator.java

```diff
@@ -96,6 +98,29 @@ public boolean isValid(UpdateResultsAPIObject updateResultsAPIObject, Constraint
                         ));
             }
 
+            // Validate container names and image names
+            List<ContainerData> kruizeContainers = kruizeObject.getKubernetes_objects().get(0).getContainerDataMap().values().stream().toList();
+            List<ContainerData> resultContainers = resultData.getKubernetes_objects().get(0).getContainerDataMap().values().stream().toList();
+            if (kruizeContainers.size() != resultContainers.size()) {
+                kubeObjsMisMatch = true;
+                errorMsg = errorMsg.concat(
+                        String.format("Containers count MisMatched. Expected: %s, Found: %s in Results for experiment: %s \n",
+                                kruizeContainers.size(), resultContainers.size(), expName));
+            } else {
+                for (int i = 0; i < kruizeContainers.size(); i++) {
+                    ContainerData kruizeContainer = kruizeContainers.get(i);
+                    ContainerData resultContainer = resultContainers.get(i);
+                    if (!kruizeContainer.getContainer_name().equals(resultContainer.getContainer_name())
+                            || !kruizeContainer.getContainer_image_name().equals(resultContainer.getContainer_image_name())) {
```

Comment:
I'd suggest that we skip the image name check for now. If the container image name has changed owing to a application upgrade, our code needs to keep track of that. We cannot penalize the updateResults API for that.

Reply:
Done!